src/overlays/actors/ovl_En_Railgibud/z_en_railgibud.c

```diff
@@ -65,122 +149,990 @@ static DamageTable D_80BA82CC = {
+        func_80BA6974(globalCtx, &sp5C, 10.0f, 10, 150, 0);
+    }
+}
+
+void func_80BA6974(GlobalContext* globalCtx, Vec3f* vec, f32 arg2, s32 arg3, s16 arg4, s16 arg5) {
+    Vec3f sp8C;
+    Vec3f sp80 = { 0.0f, 0.3f, 0.0f };
+    Vec3f sp74 = D_801D15B0;
+    s32 i;
+    s32 pad;
+
+    sp74.y = 2.5f;
+
+    for (i = arg3; i >= 0; i--) {
+        sp74.x = (Rand_ZeroOne() - 0.5f) * arg2;
+        sp74.z = (Rand_ZeroOne() - 0.5f) * arg2;
+
+        sp8C.x = vec->x + sp74.x;
+        sp8C.y = ((Rand_ZeroOne() - 0.5f) * arg2) + vec->y;
+        sp8C.z = vec->z + sp74.z;
+
+        sp74.x *= 0.5f;
+        sp74.z *= 0.5f;
+        func_800B1210(globalCtx, &sp8C, &sp74, &sp80, (s16)(Rand_ZeroOne() * arg4 * 0.2f) + arg4, arg5);
+    }
+}
+
+void func_80BA6B30(EnRailgibud* this) {
+    if (this->actor.parent == NULL) {
+        this->unk_3EC = this->unk_3EE;
+        this->unk_3EE = 1;
+        if ((this->actionFunc != func_80BA57F8) && (this->actionFunc != func_80BA66C8)) {
+            this->unk_3EE = 0;
+        }
+    } else if ((this->actionFunc != func_80BA57F8) && (this->actionFunc != func_80BA66C8)) {
+        ((EnRailgibud*)this->actor.parent)->unk_3EE = 0;
+    }
+}
+
+void func_80BA6B9C(EnRailgibud* this, GlobalContext* globalCtx) {
+    s16 temp_v0 = (this->actor.yawTowardsPlayer - this->actor.shape.rot.y) - this->unk_3E8;
+    s16 phi_a2 = CLAMP(temp_v0, -500, 500);
+
+    temp_v0 -= this->unk_3E2;
+    temp_v0 = CLAMP(temp_v0, -500, 500);
+
+    if (BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.shape.rot.y) >= 0) {
+        this->unk_3E8 += ABS_ALT(phi_a2);
+        this->unk_3E2 += ABS_ALT(temp_v0);
+    } else {
+        this->unk_3E8 -= ABS_ALT(phi_a2);
+        this->unk_3E2 -= ABS_ALT(temp_v0);
+    }
+
+    this->unk_3E8 = CLAMP(this->unk_3E8, -0x495F, 0x495F);
+    this->unk_3E2 = CLAMP(this->unk_3E2, -0x256F, 0x256F);
+}
+
+s32 func_80BA6D10(EnRailgibud* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (Player_GetMask(globalCtx) == PLAYER_MASK_GIBDO_MASK) {
+        return false;
+    }
+
+    if ((Actor_DistanceToPoint(&player->actor, &this->actor.home.pos) < 100.0f) && !(player->stateFlags1 & 0x2C6080) &&
+        !(player->stateFlags2 & 0x4080)) {
+        return true;
+    }
+
+    return false;
+}
+
+s32 func_80BA6DAC(EnRailgibud* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (Actor_DistanceToPoint(&player->actor, &this->actor.home.pos) >= 100.0f) {
+        return true;
+    }
+
+    return false;
+}
+
+void func_80BA6DF8(EnRailgibud* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->unk_3FA != 1) && (this->collider.base.acFlags & AC_HIT)) {
+        if (this->actionFunc == func_80BA57F8) {
+            this->actor.home = this->actor.world;
+        }
+        this->collider.base.acFlags &= ~AC_HIT;
+        func_800BE22C(&this->actor);
```

Comment:
```suggestion
       Actor_ApplyDamage(&this->actor);
```
The Bigpo PR changed this filename, this should be all you need to get this PR back to matching